src/components/forms/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import classnames from 'classnames'
+
+interface TextInputProps {
+  id: string
+  name: string
+  type: 'text' | 'email' | 'number' | 'password' | 'search' | 'tel' | 'url'
+  className?: string
+  error?: boolean
+  success?: boolean
+  small?: boolean
+  medium?: boolean
+  inputRef?:
+    | string
+    | ((instance: HTMLInputElement | null) => void)
+    | React.RefObject<HTMLInputElement>
+    | null
+    | undefined
+}
+
+export const TextInput = (
+  props: TextInputProps & React.InputHTMLAttributes<HTMLInputElement>
+): React.ReactElement => {
+  const {
+    id,
+    name,
+    type,
+    className,
+    error,
+    success,
+    small,
+    medium,
+    inputRef,
+    ...inputProps
+  } = props
+
+  const classes = classnames(
+    'usa-input',
+    {
+      'usa-input--error': error,
+      'usa-input--success': success,
+      'usa-input--small': small,
+      'usa-input--medium': medium,
+    },
+    className
```

Comment:
similar to the dropdown there doesn't appear to be a visual difference between disabled and enabled. I wanted to double check if that was intended or not. Also same for read only, but that may not need a visual difference either, and I don't see any details on this on the uswds site.

Reply:
same as above